### PR TITLE
Dynatrace Exporter - Fix histogram min and max calculation

### PR DESCRIPTION
**Pull request by tyrone-anz**
**Description:** 
Dynatrace exporter is rejecting histogram metrics because the exported min and max are sometimes not following `min <= avg <= max` rule.

image

The fix would set the estimated min to use average when greater than it, and max to use average when less than it.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:**

Unit test is added

**Documentation:**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@
 - `transformprocessor`: Fix issue where metric.aggregation_temporality and metric.is_monotic were not actually gettable or settable (#10197)
 - `podmanreceiver`: Container Stats Error structure (#9397)
 - `pkg/stanza`: pipeline.Operators() will return a consistently ordered list of operators whenever possible (#9761)
+- `dynatraceexporter`: Fix issue on histogram min and max calculation. (#10257)
```
Comment (dyladan) [inline CHANGELOG.md:35]: Not a very descriptive changelog entry for end users. What specifically was fixed?

```suggestion
- `dynatraceexporter`: Ensure min is always less than or equal to mean and max is always greater or equal to mean for histogram estimation. (#10257)
```

**Review by dyladan — APPROVED** [review]
Approving % changelog nit

**Review by dyladan — CHANGES_REQUESTED** [review]
~~Sorry after approving I realized that the `sum` also needs to be checked. In the case that the histogram has been seeded with negative values, the `sum` may be less than the `mean` and the upper bound of the smallest bucket, which would also fail the consistency check.~~

This is actually not correct. `min` can be greater than `sum` and it is still consistent.

**Review by dyladan — APPROVED** [review]
Approval was right the first time

**Review by arminru — APPROVED** [review]
Thank you, @tyrone-anz!

**Comment by tyrone-anz** [comment]
Thanks, everyone. Had to merge some conflict. May I get some approval to run the workflows?

**Review by codeboten — APPROVED** [review]

**Pull request merged**